lib/ansible/modules/cloud/amazon/ec2_lc.py

```diff
@@ -225,125 +325,172 @@ def create_launch_config(connection, module):
-
-    lc = LaunchConfiguration(
-        name=name,
-        image_id=image_id,
-        key_name=key_name,
-        security_groups=security_groups,
-        user_data=user_data,
-        block_device_mappings=[bdm],
-        instance_type=instance_type,
-        kernel_id=kernel_id,
-        spot_price=spot_price,
-        instance_monitoring=instance_monitoring,
-        associate_public_ip_address=assign_public_ip,
-        ramdisk_id=ramdisk_id,
-        instance_profile_name=instance_profile_name,
-        ebs_optimized=ebs_optimized,
-        classic_link_vpc_security_groups=classic_link_vpc_security_groups,
-        classic_link_vpc_id=classic_link_vpc_id,
-    )
+                bdm.update(create_block_device_meta(module, volume))
 
-    launch_configs = connection.get_all_launch_configurations(names=[name])
+    launch_configs = connection.describe_launch_configurations(
+        LaunchConfigurationNames=[name]).get('LaunchConfigurations')
     changed = False
-    if not launch_configs:
+    result = {}
+
+    launch_config['LaunchConfigurationName'] = name
+
+    if security_groups is not None:
+        launch_config['SecurityGroups'] = security_groups
+
+    if classic_link_vpc_id is not None:
+        launch_config['ClassicLinkVPCId'] = classic_link_vpc_id
+
+    if instance_monitoring:
+        launch_config['InstanceMonitoring'] = {
+            'Enabled': advanced_instance_monitoring}
+
+    if classic_link_vpc_security_groups is not None:
+        launch_config['ClassicLinkVPCSecurityGroups'] = (
+            classic_link_vpc_security_groups)
+
+    if bdm is not None:
+        launch_config['BlockDeviceMappings'] = [bdm]
+
+    if instance_profile_name is not None:
+        launch_config['IamInstanceProfile'] = instance_profile_name
+
+    if assign_public_ip is not None:
+        launch_config['AssociatePublicIpAddress'] = assign_public_ip
+
+    if user_data is not None:
+        launch_config['UserData'] = user_data
+
+    if ebs_optimized is not None:
+        launch_config['EbsOptimized'] = ebs_optimized
+
+    if len(launch_configs) == 0:
         try:
-            connection.create_launch_configuration(lc)
-            launch_configs = connection.get_all_launch_configurations(names=[name])
+            connection.create_launch_configuration(**launch_config)
+            launch_configs = connection.describe_launch_configurations(
+                LaunchConfigurationNames=[name]).get('LaunchConfigurations')
             changed = True
-        except BotoServerError as e:
+            if len(launch_configs) > 0:
+                launch_config = launch_configs[0]
+            else:
+                module.fail_json(msg="error creating launch configuration. " +
+                                 "no launch configuration returned from boto3" +
+                                 " after create.")
+        except botocore.exceptions.ClientError as e:
             module.fail_json(msg=str(e))
```

Comment:
Can you make this something like:
`module.fail_json(msg="Failed to create launch configuration", exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))`? It will be helpful for future debugging.